addons/hr_timesheet/project_timesheet.py

```diff
@@ -59,3 +60,9 @@ def _hours_get(self):
     subtask_count = fields.Integer(compute='_get_subtask_count', type='integer', string="Sub-task count")
 
     _constraints = [(osv.osv._check_recursion, 'Circular references are not permitted between tasks and sub-tasks', ['parent_id'])]
+
+    @api.constrains('parent_id')
+    def _check_subtask_project(self):
+        for task in self:
+            if task.parent_id and task.project_id and task.project_id != task.parent_id.project_id.subtask_project_id:
```

Comment:
Why ?

Only need to add a constraint that forbid to set a parent task that has no sub-task project set .